### PR TITLE
Optimize distance computation for line clustering

### PR DESCRIPTION
**Pull request by atsju**
Matrix is symetric but from my tests, you don't need the second half

### CODE REVIEW

File: word_detector/__init__.py [hunk 1]
```diff
@@ -119,6 +119,7 @@ def _cluster_lines(detections: List[DetectorRes],
             union = a.h + b.h - intersection
             iou = np.clip(intersection / union if union > 0 else 0, 0, 1)
             dist_mat[i, j] = 1 - iou  # Jaccard distance is defined as 1-iou
```
Comment (githubharald) [inline word_detector/__init__.py]: please write `dist_mat[i, j] = dist_mat[j, i] = 1 - iou` here to get the full matrix. I don't want to rely on internals of the DBSCAN implementation.

File: word_detector/__init__.py [hunk 2]
```diff
@@ -119,6 +119,7 @@ def _cluster_lines(detections: List[DetectorRes],
...
+            #dist_mat[j, i] = dist_mat[i, j] # the matrix is symetric
```
Comment (githubharald) [inline word_detector/__init__.py]: remove commented line.

**Comment by githubharald** [comment]
added comments, when the code is changed accordingly I'll merge.

File: word_detector/__init__.py (hunk 2 above)
Comment (atsju) [inline word_detector/__init__.py]:
```suggestion
            dist_mat[i, j] = dist_mat[j, i] = 1 - iou  # Jaccard distance is defined as 1-iou
```

You can use "suggestion" in reviews. I think it makes things easier to understand
image

**Comment by githubharald** [comment]
thanks for the PR, merged it.

**Pull request merged**